File: easy/2946_matrix_similarity_after_cyclic_shifts/solution.cpp
#include <vector>

class Solution {
public:
  bool areSimilar(std::vector<std::vector<int>> &mat, int k) {
    const size_t rows = mat.size();
    const size_t cols = mat[0].size();

    k %= cols;

    for (size_t row = 0; row < rows; ++row)
      for (size_t col = 0; col < cols; ++col)
        if (mat[row][col] != mat[row][(col + k) % cols])
          return false;

    return true;
  }
};

File: easy/2946_matrix_similarity_after_cyclic_shifts/solution.py
from typing import List


class Solution:
    def areSimilar(self, mat: List[List[int]], k: int) -> bool:
        rows = len(mat)
        cols = len(mat[0])

        k %= cols

        for row in range(rows):
            for col in range(cols):
                if mat[row][col] != mat[row][(col + k) % cols]:
                    return False

        return True
